p2p/node/api.go

```diff
@@ -190,8 +190,11 @@ func (p *P2PNode) requestAndWait(peerID peer.ID, topic *pubsubManager.Topic, req
 			"topic":  topic.String(),
 		}).Trace("Received data from peer")
 
-		// Mark this peer as behaving well
-		p.peerManager.MarkResponsivePeer(peerID, topic)
+		if recvd == nil {
```

Comment:
i thought we never receive a nil response, unless we will in the future somehow return nil here? and then regard it as a Nack.

Reply:
That is not how I understood it. As I understand it, today, we respond nil if we do not have the requested data. I will confirm though.

Reply:
no, its not how it works now, we send the quaiMessage back with empty response, but it will return with ID and message type